### ISSUE TITLE
CE loss error in new version

### ISSUE THREAD

**Issue by sipie800** [open-mmlab/mmsegmentation#3406]
in ce implement, line 66

            label_weights = torch.tensor([class_weight[cls] for cls in label],
                                         device=class_weight.device)

Not sure other version torch support this torch.tensor() cast. In torch 2.1, it's bound to be not supported.

**Comment by AndrewScouten** [comment]
I just ran into this error on torch 2.1 with cuda 11.8. 

I replaced it with this and it at least "works", but I don't know much if I'm being honest (about mmsegmentation, pytorch, or weighted loss 😄):

```python
label_weights = class_weight[label]
```

**Comment by AndrewScouten** [comment]
If you don't want to change this every install, [this](https://stackoverflow.com/a/72105086) was a good resource for me. With anaconda, I essentially did this:

1. Copy source into working directory

        cp $CONDA_PREFIX/lib/python3.8/site-packages/mmseg/models/losses/cross_entropy_loss.py .

2. Edit source you just copied

        label_weights = class_weight[label]
 
3. Create diff (you can remove the package change at the top & delete the copied source in your working directory after)

        diff -u $CONDA_PREFIX/lib/python3.8/site-packages/mmseg/models/losses/cross_entropy_loss.py ./cross_entropy_loss.py > ./.patches/mmseg/models/losses/cross_entropy_loss.patch
        
4. Patch after install, as an example I have a `setup.sh` that looks somewhat like this
```sh
#!/bin/sh

## System Dependencies
sudo apt-get update && \
    sudo apt-get install -y \
        python3-opencv \
        libglib2.0-0

## Create environment
conda create --name openmmlab python=3.8 -y
eval "$(conda shell.bash hook)"
conda activate openmmlab

## Install Dependencies

# ** REPLACE WITH YOUR PYTORCH VERSION BELOW **
pip install torch==2.0.1 torchvision==0.15.2 \
    --index-url https://download.pytorch.org/whl/cu118

pip install -r ./requirements/pip.txt
mim install -r ./requirements/mim.txt
pip install "mmsegmentation==1.2.1"

patch $CONDA_PREFIX/lib/python3.8/site-packages/mmseg/models/losses/cross_entropy_loss.py < \
    ./.patches/mmseg/models/losses/cross_entropy_loss.patch
```

You will have to change it depending on python version & environment system. The link at the top had something for pipenv as well.

Good luck!
